scripts/cargo-rmc

```diff
@@ -95,6 +95,12 @@ def main():
             args.verbose, dry_run=args.dry_run):
         return 1
 
+    if EXIT_CODE_SUCCESS != rmc.run_goto_instrument(
+            cbmc_filename, cbmc_filename,
+            ['--drop-unused-functions'],
```

Comment:
Is there a reason for not adding this argument to the previous `rmc.run_goto_instrument` call? What happens if you do?

Reply:
Good point. I did not see any difference in behavior on the regressions whether I added it to the previous `run_goto_instrument` call or in a separate `run_goto_instrument` call. One thing I'm unsure about though is how `--drop-unused-functions` would interact with other options specified to `goto-instrument` like `--generate-function-body-options` or `--generate-function-body` in terms of the order in which they're applied.

Reply:
Can you run a quick experiment to check how many functions are dropped in each case? I would use a project larger than the ones in our regression.

Reply:
Thanks for the suggestion. I tried on a couple of bigger tests, and there was no difference in the number of dropped functions reported by `goto-instrument`. The CBMC output had some minor differences that did not seem significant. So overall, it seems that there is no difference in behavior between applying `--drop-unused-functions` in a separate `goto-instrument` call versus applying it in the previous one. With that said, I noticed that the `goto-instrument --drop-unused-functions` call I added to the `rmc` script is incorrectly put under `if args.undefined_function_checks`, while we want it to run independent of that option. The equivalent code in `cargo-rmc` is calling `goto-instrument --add-library` and `goto-instrument --generate-function-body-options assert-false ...` unconditionally, which I think should be fixed?